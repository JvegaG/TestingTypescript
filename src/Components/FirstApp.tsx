import React from 'react'

export interface firstAppProps{
  titulo: string,
  subtitulo?: string
}


const FirstApp: React.FC<firstAppProps> = ({titulo, subtitulo}) => {
  return (
    <>
      <h1 data-testid='title'> {titulo} </h1>
      <p data-testid='subtitle0'>{subtitulo}</p>
      <p data-testid='subtitle1'>{subtitulo}</p>
    </>
  )
}

export default FirstApp
