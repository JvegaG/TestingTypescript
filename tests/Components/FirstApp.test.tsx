import React from "react"
import { getAllByText, render, screen } from '@testing-library/react'
import FirstApp from '../../src/Components/FirstApp'


describe('Testing <FirstApp />', () => {

    const title = 'Epa! Yisus'
    const subtitle = 'Vite es Rapidisimo'

    test('should create a sanpshot', () => {

        const { container } = render(
            <FirstApp
                titulo={title}
                subtitulo={subtitle}
            />
        ); // 'container' es un objeto DOM https://testing-library.com/docs/react-testing-library/api/#render


        /**
         * Se realiza una fotografia de la vista que tiene nuestro componente
         * si en un futuro se cambia esta vista accidentalmente, este test serviria
         * para saber que codigo es el que se modifico y el que se diferencia del
         * archivo .snap generado
         */
        expect(container).toMatchSnapshot();

    })

    test('should confirm that subtitle has <p></p>', () => {

        // const { container, getByText } = render(<FirstApp titulo={title} subtitulo={subtitle} />); 

        // const numSubtitulos = container.querySelectorAll('p') 
        // const firstSub = container.querySelector('p');
        // const inner = firstSub?.innerHTML

        // expect(numSubtitulos.length).toBe(2);
        // expect(getByText('CounterApp')).notw.toBeTruthy();
        // expect(inner).toContain('Vite')
        // expect(inner).toBe(`${subtitle}`)

        render(<FirstApp titulo={title} subtitulo={subtitle} />);

        const arraySubtitulos = screen.getAllByText(subtitle)

        expect(arraySubtitulos.length).toBe(2);
        expect(arraySubtitulos[0].innerHTML).toContain('Vite')
        expect(arraySubtitulos[0].innerHTML).toBe(subtitle)

    })


    test('should show the message "Epa! Yisus"', () => {
        render(<FirstApp titulo={title} />);

        expect(screen.getByText(title)).toBeTruthy();

    })


    test('should verify if exist 2 subtitle', () => {

        render(<FirstApp titulo={title} subtitulo={subtitle} />);

        expect(screen.getAllByText(subtitle).length).toBe(2);
    })

    test('should verify id exist', () => {

        render(<FirstApp titulo={title} subtitulo={subtitle} />); 

        expect(screen.getByTestId('subtitle0')).toBeTruthy();
        expect(screen.getByTestId('subtitle1').innerHTML).toBe(subtitle);

        // expect(getByTestId('test-title')).toBeTruthy();
        // expect(getByTestId('test-title').innerHTML).toBe('CounterApp');

    })
})