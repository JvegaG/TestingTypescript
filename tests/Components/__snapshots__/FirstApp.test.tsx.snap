// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <FirstApp /> should create a sanpshot 1`] = `
<div>
  <h1
    data-testid="title"
  >
     
    Epa! Yisus
     
  </h1>
  <p
    data-testid="subtitle0"
  >
    Vite es Rapidisimo
  </p>
  <p
    data-testid="subtitle1"
  >
    Vite es Rapidisimo
  </p>
</div>
`;
